lang/src/lib.rs

```diff
@@ -358,6 +361,17 @@ pub mod __private {
             input.to_bytes()
         }
     }
+
+    #[doc(hidden)]
+    pub fn _emit_cpi_invoke(ix_data: Vec<u8>, program: &AccountInfo) -> anchor_lang::Result<()> {
```

Comment:
Don't mean to nit, but what's the advantage of having this extra function vs just inling this code in the one place it's used above?

Reply:
I thought it was cleaner development, really no advantage/disadvantage that I know of off top

Reply:
Let's inline it, instead. The reasoning is that removes this __private mod hack and avoid a file switch for first time readers of the code.